spec/controllers/divisions_controller_spec.rb

```diff
@@ -1,5 +1,76 @@
 require 'spec_helper'
 
 describe DivisionsController, :type => :controller do
+  describe "#index" do
+    before :each do
+      Division.delete_all
 
+      @division = Division.create(id: 5, name: "hey jude", date: Date.new(2016,12,25),
+      number: 1, house: "representatives", source_url: "", debate_url: "", motion: "",
+      source_gid: "", debate_gid: "")
+
+      @other_division = Division.create(id: 6, name: "beautiful boy", date: Date.new(2016,06,01),
+      number: 1, house: "representatives", source_url: "", debate_url: "", motion: "",
+      source_gid: "", debate_gid: "")
+    end
+
+    context "when request has no parameters" do
+      it "should return index page with all divisions" do
+        get :index
+
+        expect(response).to render_template "divisions/index"
+        expect(response.status).to be 200
+        expect(assigns(:divisions)).to eq([@division, @other_division])
+      end
+    end
+
+    context "when request has a complete date parameter that exists in the database" do
```

Comment:
I think this wording could be clearer. Maybe "when the date parameter is a full date", it "renders the index template with only divisions that occurred on that date". This removes the mention of the database from the context, because it's not really relevant at that point.

Also I think it's better to talk about the "rendering" or "assigning" here than the "returning", because we're not actually testing what appears on the page. Returns just seems like the wrong word to me :) maybe it seems confusing with the idea of a method returning something. What do you recon?